README.md

```diff
@@ -30,25 +51,230 @@ npm install --save ipfsd-ctl
+
+- These method return a factory that exposes the `spawn` method, which allows spawning and controlling ipfs nodes
+
+
+> `daemonFactory.server` 
+
+- exposes `start` and `stop` methods to start and stop the bundled http server that is required to run the remote controller.
+
+#### Spawn nodes
+
+> Spawn either a js-ipfs or go-ipfs node through `localController` or `remoteController`
+
+`spawn([options], callback)`
+
+- `options` - is an optional object with various options and ipfs config parameters
+  - `js` bool (default false) - spawn a js or go node (default go)
+  - `init` bool (default true) - should the node be initialized
+  - `start` bool (default true) - should the node be started
+  - `repoPath` string - the repository path to use for this node, ignored if node is disposable
+  - `disposable` bool - a new repo is created and initialized for each invocation
+  - `args` - array of cmd line arguments to be passed to ipfs daemon
+  - `config` - ipfs configuration options
+  
+ - `callback` - is a function with the signature `cb(err, ipfsd)` where:
+   - `err` - is the error set if spawning the node is unsuccessful
+   - `ipfsd` - is an object with two properties: 
+     - `ctl` - an [ipfs-api](https://github.com/ipfs/js-ipfs-api) instance attached to the newly created ipfs node
+     - `ctrl` - an instance of a daemon controller object
+   
+### IPFS Client (ctl)
+
+> An instance of [ipfs-api](https://github.com/ipfs/js-ipfs-api#api)
+
+
+### IPFS Daemon Controller (ctrl)
+
+> The IPFS daemon controller that allows interacting with the spawned IPFS process
+
+#### `apiAddr` (getter) 
+
+> Get the address (multiaddr) of connected IPFS API.
+
+- returns multiaddr
+
+#### `gatewayAddr` (getter)
+
+> Get the address (multiaddr) of connected IPFS HTTP Gateway.
+
+- returns multiaddr
+
+#### `repoPath` (getter)
+
+> Get the current repo path.
+
+- returns string
+
+#### `started` (getter)
+
+>  Is the node started.
+
+- returns boolean
+ 
+#### `init (initOpts, callback)`
+
+> Initialize a repo.
+
+- initOpts (optional) - options object with the following entries
+  - keysize (default 2048) - The bit size of the identiy key.
+  - directory (default IPFS_PATH) - The location of the repo.
+  - function (Error, Node) callback - receives an instance of this Node on success or an instance of `Error` on failure
+
+
+#### `shutdown (callback)`
+
+> Delete the repo that was being used. If the node was marked as `disposable` this will be called automatically when the process is exited.
+
+- function(Error) callback
+
+#### `startDaemon (flags, callback)`
+
+> Start the daemon.
+
+- flags - Flags array to be passed to the `ipfs daemon` command.
+- function(Error, IpfsApi)} callback - function that receives an instance of `ipfs-api` on success or an instance of `Error` on failure
+
+
+#### `stopDaemon (callback)`
+
+> Stop the daemon.
+
+- function(Error) callback - function that receives an instance of `Error` on failure
+
```

Comment:
There is no documentation of a `.shutdown` nor a `.cleanup` method

Reply:
Here it is - https://github.com/ipfs/js-ipfsd-ctl/pull/176/files#diff-04c6e90faac2675aa89e2176d2eec7d8R211